datetime/datetime_test.go

```diff
@@ -52,6 +52,312 @@ func skipIfDatetimeUnsupported(t *testing.T) {
+		t.Fatalf("Unexpected result: %s, expected: %s", newdt.ToTime().String(), expected)
+	}
+}
+
+func TestDatetimeAddAdjust(t *testing.T) {
+	/*
+	 How-to test in Tarantool:
+	 > date = require("datetime")
+	 > date.parse("2012-12-31T00:00:00Z") + {month = -1, adjust = "excess"}
+	*/
+	cases := []struct {
+		year   int64
+		month  int64
+		adjust Adjust
+		date   string
+		want   string
+	}{
+		{
+			year:   0,
+			month:  1,
+			adjust: NoneAdjust,
+			date:   "2013-02-28T00:00:00Z",
+			want:   "2013-03-28 00:00:00 +0000 UTC",
+		},
+		{
+			year:   0,
+			month:  1,
+			adjust: LastAdjust,
+			date:   "2013-02-28T00:00:00Z",
+			want:   "2013-03-31 00:00:00 +0000 UTC",
+		},
+		{
+			year:   0,
+			month:  1,
+			adjust: ExcessAdjust,
+			date:   "2013-02-28T00:00:00Z",
+			want:   "2013-03-28 00:00:00 +0000 UTC",
+		},
+		{
+			year:   0,
+			month:  1,
+			adjust: NoneAdjust,
+			date:   "2013-01-31T00:00:00Z",
+			want:   "2013-02-28 00:00:00 +0000 UTC",
+		},
+		{
+			year:   0,
+			month:  1,
+			adjust: LastAdjust,
+			date:   "2013-01-31T00:00:00Z",
+			want:   "2013-02-28 00:00:00 +0000 UTC",
+		},
+		{
+			year:   0,
+			month:  1,
+			adjust: ExcessAdjust,
+			date:   "2013-01-31T00:00:00Z",
+			want:   "2013-03-03 00:00:00 +0000 UTC",
+		},
+		{
+			year:   2,
+			month:  2,
+			adjust: NoneAdjust,
+			date:   "2011-12-31T00:00:00Z",
+			want:   "2014-02-28 00:00:00 +0000 UTC",
+		},
+		{
+			year:   2,
+			month:  2,
+			adjust: LastAdjust,
+			date:   "2011-12-31T00:00:00Z",
+			want:   "2014-02-28 00:00:00 +0000 UTC",
+		},
+		{
+			year:   2,
+			month:  2,
+			adjust: ExcessAdjust,
+			date:   "2011-12-31T00:00:00Z",
+			want:   "2014-03-03 00:00:00 +0000 UTC",
+		},
+		{
+			year:   0,
+			month:  -1,
+			adjust: NoneAdjust,
+			date:   "2013-02-28T00:00:00Z",
+			want:   "2013-01-28 00:00:00 +0000 UTC",
+		},
+		{
+			year:   0,
+			month:  -1,
+			adjust: LastAdjust,
+			date:   "2013-02-28T00:00:00Z",
+			want:   "2013-01-31 00:00:00 +0000 UTC",
+		},
+		{
+			year:   0,
+			month:  -1,
+			adjust: ExcessAdjust,
+			date:   "2013-02-28T00:00:00Z",
+			want:   "2013-01-28 00:00:00 +0000 UTC",
+		},
+		{
+			year:   0,
+			month:  -1,
+			adjust: NoneAdjust,
+			date:   "2012-12-31T00:00:00Z",
+			want:   "2012-11-30 00:00:00 +0000 UTC",
+		},
+		{
+			year:   0,
+			month:  -1,
+			adjust: LastAdjust,
+			date:   "2012-12-31T00:00:00Z",
+			want:   "2012-11-30 00:00:00 +0000 UTC",
+		},
+		{
+			year:   0,
+			month:  -1,
+			adjust: ExcessAdjust,
+			date:   "2012-12-31T00:00:00Z",
+			want:   "2012-12-01 00:00:00 +0000 UTC",
+		},
+		{
+			year:   -2,
+			month:  -2,
+			adjust: NoneAdjust,
+			date:   "2011-01-31T00:00:00Z",
+			want:   "2008-11-30 00:00:00 +0000 UTC",
+		},
+		{
+			year:   -2,
+			month:  -2,
+			adjust: LastAdjust,
+			date:   "2011-12-31T00:00:00Z",
+			want:   "2009-10-31 00:00:00 +0000 UTC",
+		},
+		{
+			year:   -2,
+			month:  -2,
+			adjust: ExcessAdjust,
+			date:   "2011-12-31T00:00:00Z",
```

Comment:
It's a bit weird than `date` and `want` have different formats.

Reply:
It's a good idea, fixed!